pycbc/psd/estimate.py

```diff
@@ -65,7 +65,8 @@ def welch(timeseries, seg_len=4096, seg_stride=2048, window='hann',
     seg_stride : int
         Separation between consecutive segments, in samples.
     window : {'hann'}
-        Function used to window segments before Fourier transforming.
+        Function used to window segments before Fourier transforming, or
+        a `numpy.ndarray` that specifies the window.
```

Comment:
I'm a bit uneasy about having a single argument that can be passed either a string or a ndarray and then needs a bunch of 'suck it and see' tests to work out what to do.  Personally would prefer to have two arguments, e.g. window_name with default 'hann' and window_array with default None - that would (IMO) make the logic easier to read.  (User should set window_name explicitly to None in order to use a ndarray window.)

Reply:
@tdent, I would argue that as long as the function documentation is up to scratch, having a multi-typed argument is fine, but then again, I'm not a maintainer of this package...